core/src/main/scala/org/apache/spark/storage/BlockInfoManager.scala

```diff
@@ -421,7 +421,8 @@ private[storage] class BlockInfoManager(trackingCacheVisibility: Boolean = false
    */
   def lockNewBlockForWriting(
```

Comment:
Minor doc nit:

The `@return` docstring a couple of lines should probably be updated because the existing statement about read locks is inaccurate after your change.

I think the new condition could be something like this

> `@return` true if the block did not already exist, false otherwise. If this returns `true`, a write lock on the new block will be held. If this returns `false` then a read lock will be held if `keepReadLock == true`.